frontend/server/src/Controllers/Problem.php

```diff
@@ -5828,4 +5829,58 @@ public static function getLibinteractiveGenForSmarty(\OmegaUp\Request $r): array
             \OmegaUp\FileHandler::deleteDirRecursively($dirname);
         }
     }
+
+    /**
+     *
+     * @return array{smartyProperties: array{payload: CollectionDetailsPayload, title: \OmegaUp\TranslationString}, entrypoint: string}
+     *
+     * @omegaup-request-param string $collection_type
+     */
+    public static function getCollectionsDetailsForSmarty(\OmegaUp\Request $r): array {
+        $collectionType = $r->ensureString('collection_type');
+
+        $collection = [];
+
+        $title = new \OmegaUp\TranslationString(
+            'omegaupTitleCollectionsByLevel'
+        );
+
+        if ($collectionType !== 'author') {
+            $collection = \OmegaUp\Controllers\Tag::getFrequentTagsByLevel(
+                $r->ensureString('collection_type')
+            );
+        } else {
+            // Author
```

Comment:
aquí es mejor usar lógica positiva:

```suggestion
        if ($collectionType === 'author') {
            // L5854-5974, así ya no se necesita el comentario de L5853
        } else {
            $collection = \OmegaUp\Controllers\Tag::getFrequentTagsByLevel(
                $r->ensureString('collection_type')
            );
        }
```